puremagic/main.py

```diff
@@ -410,7 +412,24 @@ def what(file: Union[os.PathLike, str, None], h: Union[str, bytes, None]) -> Opt
     ```
     imghdr documentation: https://docs.python.org/3.12/library/imghdr.html
     imghdr source code: https://github.com/python/cpython/blob/3.12/Lib/imghdr.py
+
+    imghdr_strict enables bug-for-bug compatibility between imghdr.what() and puremagic.what() when the imghdr returns
+    a match but puremagic returns None.  We believe that imghdr is delivering a "false positive" in each of these
+    scenerios but we want puremagic.what()'s default behavior to match imghdr.what()'s false positives so we do not
+    break existing applications.
+
+    If imghdr_strict is True (the default) then a lookup will be done to deliver a matching result on all known false
+    positives.  If imghdr_strict is False then puremagic's algorithms will determine the image type.  True is more
+    compatible while False is more correct.
+
+    NOTE: This compatibility effort only deals false positives and we are not interested to track the opposite
+    situation where puremagic's deliver a match while imghdr would have returned None.  Also, puremagic.what() can
+    recognize many more file types than the twelve image file types that imghdr focused on.
     """
+    if isinstance(h, str):
+        raise TypeError("h must be bytes, not str.  Consider using bytes.fromhex(h)")
+    if h and imghdr_strict and (ext := imghdr_bug_for_bug.get(h)):
+        return ext
```

Comment:
Does this bug only exist when sending in those exact bytes, or also when those bytes are in part of a file?

Reply:
It should be that the non-passing tests are down to PureMagic's mostly superior matching rules. This should be in relation to [my comment](https://github.com/cdgriffith/puremagic/pull/76#issuecomment-2123154131) where @cclauss was testing dummy strings against the database and getting fails or no returned data.

Reply:
My tests come from reading the `test_xxx()` functions in https://github.com/python/cpython/blob/3.12/Lib/imghdr.py ?  They are very byte-by-byte tests so are quite specific.  If you think there are other positives to add to our tests, they are easy to add.